cpp/json_parser.h

```diff
@@ -21,23 +21,51 @@ inline ValueType Lookup(const picojson::object& json, const std::string& key) {
   return it->second.get<ValueType>();
 }
 
+template <typename ValueType>
+inline ValueType Lookup(const picojson::array& json, int index) {
+  CHECK(index < json.size()) << "IndexError: json::array index out of range";
+  auto value = json.at(index);
+  CHECK(value.is<ValueType>()) << "ValueError: value at index `" << index
+                               << "` has unexpected type";
+  return value.get<ValueType>();
+}
+
+tvm::runtime::DataType DataTypeFromString(const std::string& s) {
```

Comment:
methods defined in the header have to be marked as "inline" to avoid ODR violations in most of the cases